### PR TITLE
feat: add @web3-onboard/okx

### PR DESCRIPTION
**Pull request by vanishcode**
### Description

okx sdk module support

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

File: packages/okx/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          const downloadUrl =
+            device.type === 'mobile'
+              ? 'https://www.okx.com/download?deeplink=' +
+                encodeURIComponent(
+                  'okx://wallet/dapp/url?dappUrl=' +
+                    encodeURIComponent(window.location.href)
+                )
+              : 'https://www.okx.com/download'
+          window.open(downloadUrl, '_blank')
+          throw new Error('Please Install OKX Wallet before proceeding')
```
Comment (Adamj1232) [inline packages/okx/src/index.ts:32]:
```suggestion
          const downloadUrl =
            device.type === 'mobile'
              ? 'https://www.okx.com/download?deeplink=' +
                encodeURIComponent(
                  'okx://wallet/dapp/url?dappUrl=' +
                    encodeURIComponent(window.location.href)
                )
              : 'https://www.okx.com/download'
             throw new Error(
               createDownloadMessage('OKX Wallet', downloadUrl)
             )
```

`createDownloadMessage` is imported from web3-onboard/common
This provides a clean download link/message without routing the user without their consent

Comment (vanishcode) [reply packages/okx/src/index.ts:32]: 👌 Okay, I'll get right on this.

File: packages/okx/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { EIP1193Provider } from '@web3-onboard/common'
```
Comment (Adamj1232) [inline packages/okx/src/types.ts:1]:
```suggestion
import type { EIP1193Provider } from '@web3-onboard/common'
```

**Review by Adamj1232 — APPROVED** [review]
Looks great @vanishcode! Thank you again for the collaboration and the hard work!

**Pull request merged**